packages/material-ui/src/TablePagination/TablePagination.js

```diff
@@ -1,74 +1,144 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import { chainPropTypes, integerPropType } from '@material-ui/utils';
 import clsx from 'clsx';
-import withStyles from '../styles/withStyles';
+import { chainPropTypes, integerPropType } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses, isHostComponent } from '@material-ui/unstyled';
+import experimentalStyled, { shouldForwardProp } from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
 import InputBase from '../InputBase';
 import MenuItem from '../MenuItem';
 import Select from '../Select';
 import TableCell from '../TableCell';
 import Toolbar from '../Toolbar';
 import Typography from '../Typography';
-import TablePaginationActions from './TablePaginationActions';
+import PaginationActions from './TablePaginationActions';
 import useId from '../utils/useId';
+import { getTablePaginationUtilityClass } from './tablePaginationClasses';
+
+const makeOverridesResolver = (slotName) => {
+  return function overridesResolver(_, styles) {
+    return styles[slotName];
+  };
+};
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    color: theme.palette.text.primary,
-    fontSize: theme.typography.pxToRem(14),
-    overflow: 'auto',
-    // Increase the specificity to override TableCell.
-    '&:last-child': {
-      padding: 0,
-    },
+const TablePaginationRoot = experimentalStyled(
+  TableCell,
+  { shouldForwardProp: (prop) => shouldForwardProp(prop) || prop === 'classes' },
```

Comment:
Diverge from standard

```suggestion
  {},
```